marshal/consumer.go

```diff
@@ -458,76 +485,59 @@ func (c *Consumer) claimTopics() {
 	}
 }
 
-// updatesTopicClaims if changed. It should be noted that it expects c.lock.RLock
-// to be already acquired
-func (c *Consumer) safeUpdateTopicClaims(latestClaims map[string]bool, force bool) {
```

Comment:
I don't think that's a valid constraint. It doesn't seem a reasonable design that you can't the same map twice -- the channel is just saying 'hi, here's my list of claimed topics'?